heat/core/dndarray.py

```diff
@@ -582,15 +582,15 @@ def create_lshape_map(self, force_check: bool = False) -> torch.Tensor:
             result. Otherwise, create the lshape_map
         """
         if not force_check and self.__lshape_map is not None:
-            return self.__lshape_map
+            return self.__lshape_map.clone()
```

Comment:
I can see why you do it, but the lshape_map can get quite big with many nodes. Why not cloning or copying as need arises?

Reply:
Well somehow it must be ensured that the attribute is immutable, and as far as I know, there are no read-only Tensors.

Reply:
most of the time this shouldnt be a problem. the `lshape_map` is generally much smaller than the other tensor sizes